src/App.tsx

```diff
@@ -14,48 +15,106 @@ export const goodsFromServer = [
   'Jam',
   'Garlic',
 ];
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+  type ReorderOptions = {
+    sortType: SortType,
+    isReversed: boolean,
+  };
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((a, b) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return a.localeCompare(b);
+      case SortType.LENGTH:
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    return visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
 
 export const App: React.FC = () => {
+  const [sortType, setSortType] = useState(SortType.NONE);
+  const [isReversed, setIsReversed] = useState(false);
+
+  const reverse = () => {
+    setIsReversed(true);
+  };
```

Comment:
What would be if i click reverse 2 times? List must change to original order. It doesn't work.
Failed test said you same info. Please don't push solutions with failed tests